src/backend/replication/logical/message.c

```diff
@@ -93,3 +96,62 @@ logicalmsg_redo(XLogReaderState *record)
 
 	/* This is only interesting for logical decoding, see decode.c. */
 }
+
+/*
+ * NEON: persist file in WAL to save it in persistent storage.
+ * If fd < 0, then remote entry from page server.
+ */
+void
+wallog_file_descriptor(char const* path, int fd, uint64_t limit)
```

Comment:
```suggestion
/*
 * NEON: persist file in WAL to save it in persistent storage.
 * If fd < 0, then remove entry from page server.
 */
void
wallog_file_descriptor(char const* path, int fd, uint64_t limit)
```

This is a dangerous API. Let's have two separate functions, `wallog_file_descriptor` and `wallog_file_removal`.

Reply:
Done